### PR TITLE
dev/core#1532 - Remove civicaseActivityRevisions setting & related code

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Removes an old deprecated CiviCase setting. Removes some related code that no longer matters. Mark the fields it pertains to as deprecated.
This is a followup from #19422 which flagged the setting but did not remove it.

See discussion at https://lab.civicrm.org/dev/core/-/issues/1532

Technical Details
----------------------------------------
This setting has been deprecated for years, and core CiviCase doesn't really respect it anyway. Only the API was really paying attention to it. Time to get rid of that stuff as a preamble to ripping out all of the case activity revisioning.

### CODE REVIEW

**Comment by demeritcowboy** [comment]
x-ref: https://lab.civicrm.org/dev/core/-/issues/1532

File: api/v3/Activity.php [hunk 1]
```diff
@@ -63,71 +63,14 @@ function civicrm_api3_activity_create($params) {
...
 
   if (isset($activityBAO->id)) {
-    if ($case_id && $isNew && !$createRevision) {
+    // Fixme - Move business logic out of API
+    if (!empty($params['case_id']) && $isNew) {
       // If this is a brand new case activity, add to case(s)
-      foreach ((array) $case_id as $singleCaseId) {
+      foreach ((array) $params['case_id'] as $singleCaseId) {
         $caseActivityParams = ['activity_id' => $activityBAO->id, 'case_id' => $singleCaseId];
         CRM_Case_BAO_Case::processCaseActivity($caseActivityParams);
```
Comment (colemanw) [inline api/v3/Activity.php:75]: This right here is the motivation for this PR. I need to get `$params['case_id']` working in APIv4 `Activity.create` and the best solution is to move this logic to the BAO.
See #24244

**Comment by demeritcowboy** [comment]
I think this is mergeable my only question is about communication since as of this PR now api will also stop making revisions, and in the lab ticket (albeit 2 yrs ago) it was noted that drupal views/webform might rely on it for something (@mattwire). The PR doesn't remove the setting or fields from the db - getting the setting's value still returns "1" and the old revisions are still there - just creating a new activity via api won't create a revision.

**Comment by demeritcowboy** [comment]
Have added upgrade warning message: https://github.com/civicrm/civicrm-core/pull/24266

**Pull request merged**